### PR TITLE
[Enhancement] support BIGINT/INT argument for window_funnel

### PR DESCRIPTION
**Pull request by Pslydhh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

- Reduce copy when access array daum.
- Add BIGINT and INT as argument of window_funnel to remove cast costs.

### CODE REVIEW

File: be/src/exprs/agg/window_funnel.h [hunk 1]
```diff
@@ -33,12 +33,38 @@ struct ComparePairFirst final {
 
 enum FunnelMode : int { DEDUPLICATION = 1, FIXED = 2, DEDUPLICATION_FIXED = 3 };
 
+namespace InteralTypeOfFunnel {
+template <PrimitiveType primitive_type>
+struct TypeTraits {};
+
+template <>
+struct TypeTraits<TYPE_INT> {
```
Comment (mofeiatwork) [inline be/src/exprs/agg/window_funnel.h:41]: What's the purpose? What's the difference of `RunTimeTypeTraits` ?

Comment (Pslydhh) [reply be/src/exprs/agg/window_funnel.h:41]: > What's the purpose? What's the difference of `RunTimeTypeTraits` ?

For DATE and DATETIME we process internal type of CppType(DateValue and TimestampValue), for BIGINT and INT just CppType, so we should extract concrete type first.

**Comment by kangkaisen** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — DISMISSED** [review]

**Review by choury — DISMISSED** [review]
lgtm

**Comment by stdpain** [comment]
LGTM, can you add a case for how to using window_funnel(int, int, bool[])

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|4|4|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**